### ISSUE TITLE
testAbnormalTermination should not use 'false' command

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3149]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#3149|
|Type|defect|
|Created|2008-04-02 15:37:51Z|
|Branch|https://github.com/twisted/twisted/tree/replace-false-3149|

Some crazy platforms (solaris) make it with a 255 return code instead of 1. A simple 'python -c "import sys; sys.exit(1)"' should do the trick.

<details><summary>Searchable metadata</summary>

```
trac-id__3149 3149
type__defect defect
reporter__therve therve
priority__low low
milestone__ 
branch__branches_replace_false_3149 branches/replace-false-3149
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1207150671000000 1207150671000000
changetime__1208690239000000 1208690239000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [23301]) Branching to 'replace-false-3149'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [23302]) Replace the call to false with a python call.

Refs [#3149](https://github.com/twisted/twisted/issues/3149)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

Problem fixed (see http://buildbot.twistedmatrix.com/builders/solaris-py2.4-select/builds/7/steps/select/logs/problems). Please review!

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set owner to @therve|
|-|-|

Looks good to me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [23338]) Merge replace-false-3149

Author: therve
Reviewer: ralphm
Fixes [#3149](https://github.com/twisted/twisted/issues/3149)

Replace the use of /bin/false in test_process by a python call, so that exit
code is more controlled. This fixes 2 tests under Solaris.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
